### PR TITLE
Improve coordinate checks

### PR DESCRIPTION
**Pull request by hayfield**
This makes several improvements to better check whether a coordinate is valid.

* Fixes an exception identified in #95 
* Deems out-of-range coordinates to be invalid
* Maintains assertion that a `(0,0)` coordinate is not a correct coordinate, despite being valid against the IATI Standard (it's in the ocean and so almost certainly an incorrect value)

Tests have been added.

### CODE REVIEW

**Comment by dalepotter** [comment]
Tests failed :-(

File: stats/dashboard.py [hunk 1]
```diff
@@ -243,13 +243,18 @@ def valid_value(value_element):
 
 
 def valid_coords(x):
```
Comment (dalepotter) [inline stats/dashboard.py]: Could we make life easier for ourselves in future by documenting this function to our better practices...

Suggest adding a docstring here

File: stats/dashboard.py [hunk 2]
```diff
@@ -243,13 +243,18 @@ def valid_value(value_element):
...
+        return False
     if len(coords) != 2:
         return False
     try:
         x = decimal.Decimal(coords[0])
         y = decimal.Decimal(coords[1])
-        if x == 0 and y ==0:
+        if x == 0 and y == 0:
+            return False
+        elif x < -90 or x > 90 or y < -180 or y > 180:
```
Comment (dalepotter) [inline stats/dashboard.py]: Also, on the documentation front: might be worth adding a one-line comment to say what this is doing...

**Comment by hayfield** [comment]
Documentation added.

Also changed coordinate naming to clarify that we're talking `(lat, lng)` rather than a generic coordinate space.

**Pull request merged**